### ISSUE TITLE
Replace back to back sharded2interleaved and interleaved2sharded in SD with reshard

### ISSUE THREAD

**Issue by AleksKnezevic** [tenstorrent-metal/tt-metal#6182]
Please replace all the back to back conversions to interleaved and back to sharded with reshard. 

In order to do this safely, we should first collect all the conversion shard specs, and create unit tests for reshard, and once those unit tests pass, make the modification in the model.

**Comment by AleksKnezevic** [comment]
@mtatsumiTT, can you please update with progress?

**Comment by mtatsumiTT** [comment]
I collected input/output configs for reshard op in SD model and tested, but it's currently blocked either 1KB issue (#6206) or shard-spec error as following:
```
E       RuntimeError: TT_FATAL @ tt_eager/tt_dnn/op_library/sharded/sharded_op.cpp:91: same_row_size
E       info:
E       row major must have shard_spec[1] be the same on both input and output
```
However, the shard-spec of conv2d ops would likely change after @tt-nshanker 's conv2d 2.0 so we have not asked for specific update for reshard op yet.

**Comment by nsmithtt** [comment]
@AleksKnezevic, this is complete right?

**Comment by AleksKnezevic** [comment]
No, we still have some unsupported reshards that @tarafdarTT is working on.

**Comment by mywoodstock** [comment]
@esmalTT Is this still valid? Or can we close?

**Comment by esmalTT** [comment]
@s-jovic Assigning you but feel free to close if no longer relevant

**Comment by pavlejosipovic** [comment]
Closing as it's no longer relevant.
